_posts/2022-02-04-rust-9.0.0.md

```diff
@@ -0,0 +1,141 @@
+---
+layout: post
+title: "Recent Rust Apache Arrow and Parquet Highlights"
+date: "2022-02-04 00:00:00 -0600"
+author: pmc
+categories: [release]
+---
+<!--
+{% comment %}
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to you under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+The Rust implementation of [Apache Arrow] has just released version `9.0.0`.
+
+While a major version of this magnitude may shock some in the Rust
+community to whom it implies a slow moving 20 year old piece of
+software, nothing could be further from the truth!
+
+With regular and predictable bi-weekly releases, the library continues
+to evolve and `9.0.0` is no exception. Some recent highlights
+
+
+# `parquet`: async, performance, safety and nested types
+
+The parquet `9.0.0` release includes an `async` reader (TODO link to rustdoc
+when published), a long time requested feature. Using the `async`
+reader it is now possible to read only the relevant parts of a parquet
+file from a networked source such as object storage. Previously the
+entire file had to be buffered locally. We are hoping to add a `async`
+writer in a future release and would love some
+[help](https://github.com/apache/arrow-rs/issues/1269).
+
+It is also significantly faster to read parquet data (up to
+[60x](https://github.com/apache/arrow-rs/pull/1180#issuecomment-1018518863)
+in some cases) than with previous versions of the `parquet`
+crate. Kudos to [tustvold](https://github.com/tustvold) and
+[yordan-pavlov](https://github.com/yordan-pavlov) for their
+contributions in these areas.
```

Comment:
This is of course well outside the scope of this blogpost, so not at all suggesting that it needs to be added here. But I wanted to draw your attention to https://lists.apache.org/thread/n98tr38moqvbf1rlqd7n7912mfrqv8ps in case you missed it. I know there was some desire to get Rust benchmarks running in conbench and with Elena's work there that _should_ be possible now (just need to add a machine to run them + a PR to that repo). Elena and I are happy to help with parts of that if we can. 

On the R and Python libraries, we've found it super useful for tracking benchmarks like these (and being able to link to them publicly)

Reply:
As a starting point, I am hoping to add 8 Raspberry Pi nodes shortly. I am considering adding some more powerful hardware as well, perhaps a Mac Mini.

Reply:
That's very exciting!

Reply:
Filed https://github.com/apache/arrow-rs/issues/1274 to track this work. Thank you @jonkeane